yt_dlp/extractor/dlf.py

```diff
@@ -0,0 +1,209 @@
+import re
+from .common import InfoExtractor
+from ..utils import (
+    determine_ext,
+    extract_attributes,
+    int_or_none,
+    url_or_none,
+)
+
+
+class DLFBaseIE(InfoExtractor):
+    _VALID_URL_BASE = r'https?://(?:www\.)?deutschlandfunk\.de/'
+    _BUTTON_REGEX = r'(<button[^>]+alt="Anhören"[^>]+data-audio-diraid[^>]*>)'
+
+    def _parse_button_attrs(self, button, id=None):
+        attrs = extract_attributes(button)
+        id = id or attrs['data-audio-diraid']
+
+        # Note: DLF's mp3 recordings are in two distinct locations on the web:
+        # A direct download location, https://download.deutschlandfunk.de/ ... .mp3
+        # A stream with playback controls, https://ondemand-mp3.dradio.de/ ... .mp3
+        url = url_or_none(
+            attrs.get('data-audio-download-src')
+            or attrs.get('data-audio')
+            or attrs.get('data-audioreference')
+            or attrs['data-audio-src'])
+
+        return {
+            'id': id,
+            'title': (
+                attrs.get('data-audiotitle')
+                or attrs.get('data-audio-title')
+                or attrs.get('data-audio-download-tracking-title')),
+            'url': url,
+            'formats': (
+                self._extract_m3u8_formats(url, id) if determine_ext(url) == 'm3u8'
+                else None),
+            'duration': int_or_none(
+                attrs.get('data-audioduration')
+                or attrs.get('data-audio-duration')),
+            'thumbnail': attrs.get('data-audioimage'),
+            'uploader': attrs.get('data-audio-producer'),
+            'series': attrs.get('data-audio-series'),
+            'channel': attrs.get('data-audio-origin-site-name'),
```

Comment:
- We need to add `'vcodec': 'none'` if it's audio-only.
 - Using `formats` for both m3u8 and mp3 formats seems simpler to me
 - `traverse_obj` makes this less grueling to read/write
 - Like how we added `extractor` and `extractor_key` above, we want to extract the `webpage_url` for individual media here too:
```suggestion
            **traverse_obj(attrs, {
                'title': (('data-audiotitle', 'data-audio-title', 'data-audio-download-tracking-title'), {str}),
                'duration': (('data-audioduration', 'data-audio-duration'), {int_or_none}),
                'thumbnail': ('data-audioimage', {url_or_none}),
                'uploader': 'data-audio-producer',
                'series': 'data-audio-series',
                'channel': 'data-audio-origin-site-name',
                'webpage_url': 'data-audio-download-tracking-path',
            }, get_all=False),
            'formats': (self._extract_m3u8_formats(url, id_, fatal=False)
                        if ext == 'm3u8' else [{'url': url, 'ext': ext, 'vcodec': 'none'}])
```

Reply:
Does the `webpage_url` field also require the `url_or_none` expected type?

```python
'webpage_url': ('data-audio-download-tracking-path', {url_or_none})
```

I included this in 93f346b187ce6d5d74ca85b7f373c56cea1bb379, but I'll remove it if it's unnecessary.

Reply:
Good idea